docs/site/LB3-vs-LB4-request-response-cycle.md

```diff
@@ -0,0 +1,96 @@
+In LoopBack 4, you cannot do it anymore. The closest to doing this is to use the
+[RestApplication.mountExpressRouter()](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.mountexpressrouter.html)
+API. This API allows you to mount a custom Express router on the LoopBack app.
+
+Using [Controllers](https://loopback.io/doc/en/lb4/Controllers.html) is the
+recommended way for creating custom (and REST) endpoints on your app. Its
+support for
+[dependency injection](https://loopback.io/doc/en/lb4/Dependency-injection.html)
+and [Interceptors](https://loopback.io/doc/en/lb4/Interceptors.html) makes it a
+very powerful extension mechanism.
+
+In LoopBack 4
+[middleware.json](https://loopback.io/doc/en/lb3/middleware.json.html)
+is not require anymore because of architectural changes.
+
+In LoopBack 3, models files automatically create the corresponding REST API
+endpoints and the database query machinery (using the configured datasource).
+In LoopBack 4, model files are limited only to describing the properties of the
+data. You will have to create a corresponding
+[Repository](https://loopback.io/doc/en/lb4/Repositories.html)
+for database connectivity, and controller creating the REST API endpoint.
+
+The fact that you have to create two more artifacts along with the model to
+get a REST endpoint working might seem overly tedious at first. However, the
+separation of concerns and decoupling the functionality makes the codebase
+cleaner, easier to maintain, and much easier to customize functionality at
+various levels. This can be better appreciated as the complexity of your app
+grows.
+
+For those who are uncomfortable with the concept of having to creating a
+repository and a controller for a model, we have a component
+[@loopback/rest-crud
+](https://loopback.io/doc/en/lb4/Creating-crud-rest-apis.html)
+; with a little bit of configuration, a model file is all you will need to
+create the REST endpoints. Once your requirements outgrow what
+`@loopback/rest-crud` provides, you can implement your REST endpoints the
+idiomatic way.
+
+Components are still supported in LoopBack 4, but the concept of component
+has completely changed.
+
+In LoopBack 3, a
+[component](https://loopback.io/doc/en/lb3/LoopBack-components.html)
+is a simple Node.js module which exports a function with the signature
+`function(app, options)`. In LoopBack 4, a
+[component](https://loopback.io/doc/en/lb4/Creating-components.html)
+is a TypeScript class which can add
+[servers](https://loopback.io/doc/en/lb4/Server.html),
+[observers](https://loopback.io/doc/en/lb4/Life-cycle.html),
+[providers](https://loopback.io/doc/en/lb4/Creating-components.html#providers),
+and controllers to the app using dependency injection.
+
+A components from LoopBack 3 can be migrated to LoopBack 4 by moving the
+functionality to the controller of a LoopBack 4 component.
+
+Because of the architectural changes, `component-config.json` is not required
+in LoopBack 4 anymore.
```

Comment:
I don't see how is this relevant to request/response life cycle? IMO, this belongs to https://loopback.io/doc/en/lb4/migration-extensions.html

Reply:
Imagine someone wrote LB3 components to add routes to the app. It would help them to know what happens to those components from the context of the changes in the req-res cycle.